contracts/starknet/src/tests/test_escrow_claim.cairo

```diff
@@ -0,0 +1,339 @@
+    };
+
+    use yab::tests::utils::{
+        constants::EscrowConstants::{
+            USER, OWNER, MM_STARKNET, MM_ETHEREUM, ETH_TRANSFER_CONTRACT, ETH_USER, ETH_USER_2, ETH_USER_3
+        },
+    };
+
+    use openzeppelin::{
+        upgrades::{
+            UpgradeableComponent,
+            interface::{IUpgradeable, IUpgradeableDispatcher, IUpgradeableDispatcherTrait}
+        },
+    };
+
+    fn setup() -> (IEscrowDispatcher, IERC20Dispatcher) {
+        setup_general(BoundedInt::max(), BoundedInt::max())
+    }
+
+    // fn setup_approved(approved: u256) -> (IEscrowDispatcher, IERC20Dispatcher){
+    //     setup_general(BoundedInt::max(), approved)
+    // }
+
+    // fn setup_balance(balance: u256) -> (IEscrowDispatcher, IERC20Dispatcher){
+    //     setup_general(balance, BoundedInt::max())
+    // }
+
+    fn setup_general(balance: u256, approved: u256) -> (IEscrowDispatcher, IERC20Dispatcher){
+        let eth_token = deploy_erc20('ETH', '$ETH', BoundedInt::max(), OWNER());
+        let escrow = deploy_escrow(
+            OWNER(),
+            ETH_TRANSFER_CONTRACT(),
+            MM_ETHEREUM(),
+            MM_STARKNET(),
+            eth_token.contract_address
+        );
+
+        start_prank(CheatTarget::One(eth_token.contract_address), OWNER());
+        eth_token.transfer(USER(), balance);
+        stop_prank(CheatTarget::One(eth_token.contract_address));
+
+        start_prank(CheatTarget::One(eth_token.contract_address), USER());
+        eth_token.approve(escrow.contract_address, approved);
+        stop_prank(CheatTarget::One(eth_token.contract_address));
+
+        (escrow, eth_token)
+    }
+
+    fn deploy_escrow(
+        escrow_owner: ContractAddress,
+        eth_transfer_contract: EthAddress,
+        mm_ethereum_contract: EthAddress,
+        mm_starknet_contract: ContractAddress,
+        native_token_eth_starknet: ContractAddress
+    ) -> IEscrowDispatcher {
+        let escrow = declare('Escrow');
+        let mut calldata: Array<felt252> = ArrayTrait::new();
+        calldata.append(escrow_owner.into());
+        calldata.append(eth_transfer_contract.into());
+        calldata.append(mm_ethereum_contract.into());
+        calldata.append(mm_starknet_contract.into());
+        calldata.append(native_token_eth_starknet.into());
+        let address = escrow.deploy(@calldata).unwrap();
+        return IEscrowDispatcher { contract_address: address };
+    }
+
+    fn deploy_erc20(
+        name: felt252, symbol: felt252, initial_supply: u256, recipent: ContractAddress
+    ) -> IERC20Dispatcher {
+        let erc20 = declare('ERC20');
+        let mut calldata = array![name, symbol];
+        Serde::serialize(@initial_supply, ref calldata);
+        calldata.append(recipent.into());
+        let address = erc20.deploy(@calldata).unwrap();
+        return IERC20Dispatcher { contract_address: address };
+    }
+
+    #[test]
+    fn test_happy_path() {
```

Comment:
rename this test

Reply:
renamed on e76e5e4abfc0c95729731138ee18df718cceda40